### PR TITLE
Server extension update

### PR DESCRIPTION
**Pull request by mattbsox**

### CODE REVIEW

File: src/integTest/resources/sample.servlet/verifyTimeoutSuccessAppsTest.gradle [hunk 1]
```diff
@@ -34,7 +34,7 @@ ext {
     testServerHttpsPort = 9443
 
     // This is set in the ibm-web-ext.xml file
-    warContext = 'myLibertyApp'
+    warContext = 'myLibertyApp1'
```
Comment (cthigh) [inline src/integTest/resources/sample.servlet/verifyTimeoutSuccessAppsTest.gradle]: I remember that myLibertyApp1 was set in server.xml, but I don't see why this had to change now?  I.e., what is really different?

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy [hunk 2]
```diff
@@ -45,7 +47,12 @@ class Liberty implements Plugin<Project> {
         project.configurations.create('libertyLicense')
         project.configurations.create('libertyRuntime')
 
-        def currentServer = project.liberty.server
+        project.afterEvaluate{
+            if (project.liberty.server == null) {
+                project.liberty.server = copyProperties(project.liberty)
+            }
+            setServersForTasks(project)
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: Add a comment before this method explaining why you have to set the server object on the tasks.

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy [hunk 3]
```diff
@@ -55,92 +62,114 @@ class Liberty implements Plugin<Project> {
...
             description 'Checks if the Liberty server is running.'
             logging.level = LogLevel.INFO
-            server = currentServer
+            server = project.liberty.server
         }
 
         project.task('libertyCreate', type: CreateTask, dependsOn: 'installLiberty') {
             description 'Creates a WebSphere Liberty Profile server.'
-            outputs.file { new File(getUserDir(project), "servers/${project.liberty.serverName}/server.xml") }
+            outputs.file { new File(getUserDir(project), "servers/${project.liberty.server.name}/server.xml") }
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: Won't this line be problematic for the old style liberty closure where the object gets set in the afterEvaluate method?  In that case will outputs.file get set to "servers/null/server.xml which will never exist?

Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: Can we determine the server name for all combinations and have that in a local variable?

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy [hunk 4]
```diff
@@ -55,92 +62,114 @@ class Liberty implements Plugin<Project> {
...
+            }
         }
+        serverMap.remove('class')
+
+        return ServerExtension.newInstance(serverMap)
     }
 
+    void setServersForTasks(Project project){
+        project.tasks.withType(AbstractServerTask).each {task ->
+            task.server = project.liberty.server
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: This is going to be problematic when we introduce the named domain objects.  Without some changes here all our named server tasks will end up with the same default server object.  Maybe just add a TODO for now that we need to beware.

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/AbstractTask.groovy [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright IBM Corporation 2014, 2017.
+ * (C) Copyright IBM Corporation 2017.
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/AbstractTask.groovy]: Actually, this date should stay 2014, 2017.  This is the old class.

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy (hunk 3 above)
Comment (mattbsox) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: Ya, this is an issue. We'll need to set this after the project has been evaluated.

**Pull request merged**